docs/components/graph_system.md

```diff
@@ -0,0 +1,263 @@
+# Neo4j Graph System
+
+The graph system provides functionality for creating and managing (:Claim) and (:Sentence) nodes in the ClarifAI Neo4j knowledge graph.
+
+## Overview
+
+The graph management module provides:
+
+- **Data Models**: `Claim`, `Sentence`, `ClaimInput`, `SentenceInput` for structured node data
+- **Neo4j Manager**: `Neo4jGraphManager` for database operations 
+- **Schema Management**: Automatic constraint and index creation
+- **Batch Operations**: Efficient batch creation using Cypher UNWIND
+- **Relationship Management**: Automatic `ORIGINATES_FROM` relationships to Block nodes
+
+## Architecture
+
+This implementation follows the architectural guidelines from `docs/arch/idea-neo4J-ineteraction.md`, using **Neo4j Python Driver (Direct Cypher)** for:
+
+- Precise control over node properties including nullable evaluation scores
+- Batch operations with UNWIND for performance
+- Complex property types and relationships
+- Schema management with constraints and indexes
+
+## Data Models
+
+### ClaimInput
+
+Input data for creating Claim nodes:
+
+```python
+from clarifai_shared.graph import ClaimInput
+
+claim_input = ClaimInput(
+    text="The Earth orbits the Sun",
+    block_id="block_conversation_123_chunk_5",
+    entailed_score=0.95,
+    coverage_score=0.88,
+    decontextualization_score=0.92
+)
+```
+
+Properties:
+- `text`: The claim text content
+- `block_id`: ID of the originating Block node 
+- `entailed_score`: Optional float (0.0-1.0) - NLI entailment score
+- `coverage_score`: Optional float (0.0-1.0) - Information completeness score
+- `decontextualization_score`: Optional float (0.0-1.0) - Context independence score
+- `claim_id`: Optional string - Auto-generated if not provided
+
+### SentenceInput
+
+Input data for creating Sentence nodes:
+
+```python
+from clarifai_shared.graph import SentenceInput
+
+sentence_input = SentenceInput(
+    text="This statement is ambiguous and unclear",
+    block_id="block_conversation_123_chunk_8", 
+    ambiguous=True,
+    verifiable=False
+)
+```
+
+Properties:
+- `text`: The sentence text content
+- `block_id`: ID of the originating Block node
+- `ambiguous`: Optional boolean - Whether the sentence is ambiguous
+- `verifiable`: Optional boolean - Whether the sentence is verifiable
+- `sentence_id`: Optional string - Auto-generated if not provided
+
+## Basic Usage
+
+### Setup
+
+```python
+from clarifai_shared.graph import Neo4jGraphManager, ClaimInput, SentenceInput
+from clarifai_shared.config import load_config
+
+# Load configuration
+config = load_config()
+
+# Create graph manager
+with Neo4jGraphManager(config) as graph:
+    # Setup schema (constraints and indexes)
+    graph.setup_schema()
+    
+    # Now ready for operations...
+```
```

Comment:
isn't this "Basic Usage" just repeating what is in the tutorial? like the docs/components/import_system.md document, I would prefer a reference to the tutorial instead of mixing responsibilities in the documentation like this.

If content from this section is not yet covered in the tutorial, it should be expanded to cover it.